tools/get_tcwind_csv.py

```diff
@@ -87,42 +90,109 @@ def _get_hazard(all_lats, all_lons, location_ids=None,
 
-    # Do a spatial join to attach the query locations and location names
-    tmp = {'lat': all_lats, 'lon': all_lons}
+        # Fix up some additional columns
+        if product.lower() == 'DeepCyc':
+            df['scenario'] = ret['header']['scenario']
+            df['time_horizon'] = ret['header']['time_horizon']
+            df['simulation_years'] = ret['header']['simulation_years']
+
+        df.rename(columns={'wind_speed': 'wind_speed_{}'.format(ret['header']['wind_speed_units'])}, inplace=True)
+        all_query_dfs.append(df)
+
+    df = pd.concat(all_query_dfs, ignore_index=True)
+
+    return df
+
+
+def _do_queries_in_parallel(all_lats, all_lons,
+                             terrain_correction,
+                             wind_speed_averaging_period,
+                             product, scenario,
+                             time_horizon, return_period):
+
+    import multiprocessing as mp
+    from itertools import repeat
+
+    num_calls = ceil(len(all_lats) / 1000)
+    lats = np.array_split(all_lats, num_calls)
+    lons = np.array_split(all_lons, num_calls)
+
+    with mp.Pool(8) as pool:
+        dfs = pool.starmap(_do_queries_serially,
+                            zip(lats, lons,
+                               repeat(terrain_correction),
+                               repeat(wind_speed_averaging_period),
+                               repeat(product),
+                               repeat(scenario),
+                               repeat(time_horizon),
+                               repeat(return_period)))
+
+    df = pd.concat(dfs, ignore_index=True)
+    return df
+
+
+def _get_hazard(all_lats, all_lons, location_ids=None,
+                terrain_correction='full_terrain_gust',
+                wind_speed_averaging_period='3_seconds',
+                product='deepcyc', scenario='current_climate',
+                time_horizon='now', return_period=None):
+
+    assert len(all_lats) == len(all_lons), 'Mismatching number of lats and lons'
     if location_ids is not None:
-        tmp['location_id'] = location_ids
+        assert len(all_lats) == len(location_ids)
 
-    df_locs = gpd.GeoDataFrame(tmp, geometry=gpd.points_from_xy(all_lons, all_lats), crs="EPSG:4326")
-    df = df.set_crs(4326).sjoin(df_locs, predicate="contains")
-    df.drop(['index_right'], axis=1, inplace=True)
+    convert_to_10_minute = False
+    if wind_speed_averaging_period == '10_minutes':
+        convert_to_10_minute = True
+        wind_speed_averaging_period = '1_minute'
+        assert terrain_correction == 'open_water', \
+            '10 minutes wind speed averaging period only supported for Open Water terrain type.'
 
-    # Fix up some additional columns
-    if m.product == 'DeepCyc':
-        df['scenario'] = ret['header']['scenario']
-        df['time_horizon'] = ret['header']['time_horizon']
-        df['simulation_years'] = ret['header']['simulation_years']
+    do_parallel = True
```

Comment:
This is more of a comment than a request for changes.
Instead of having serial/parallel branches, consider splitting up the work into the correct grain, push it onto a queue, and then let nworkers determine whether its consumed in parallel or serial.


I am more used to seeing something like a work queue, where the work is split up into the correct grain, and serial/parallel depends on how many workers you have. Some optimization for n